solutions/src/main/java/ru/sberbank/school/task11/Streams.java

```diff
@@ -0,0 +1,158 @@
+
+import lombok.NonNull;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Облегченная версия класса {@link Stream}
+ */
+public class Streams<T> {
+
+    private final List<Supplier<List<T>>> suppliers = new ArrayList<>();
+    private List<T> elements = new ArrayList<>();
+
+    private Streams(Collection<T> elements) {
+        this.elements.addAll(elements);
+    }
+
+    /**
+     * Принимает на вход массив элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> Streams<T> of(@NonNull T... elements) {
+        return new Streams<>(new ArrayList<>(Arrays.asList(elements)));
+    }
+
+    /**
+     * Принимает на вход коллекцию элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> Streams<T> of(@NonNull Collection<T> elements) {
+        return new Streams<>(new ArrayList<>(elements));
+    }
+
+    /**
+     * Фильтрация элементов по заданному правилу. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Иными словами, этот метод оставляет в коллекции только те элементы, которые удовлетворяют условию в лямбде.
+     * <p>
+     * Аналог Stream#map(Function).
+     * <p>
+     * Intermediate операция.
+     *
+     * @param object - правило фильтрации элементов.
+     * @return стрим
+     */
+    public Streams<T> filter(Predicate<? super T> object) {
+        suppliers.add(() -> elements.stream()
+                .filter(object)
+                .collect(Collectors.toList()));
+        return this;
+    }
+
+    /**
+     * Преобразование элементов в какие-то другие элементы. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Например, данный метод должен уметь извлекать какие-то поля из объектов исходного типа (если это возможно).
+     * <p>
+     * Аналог Stream#map.
+     * <p>
+     * Intermediate операция.
+     *
+     * @param object - правило траснформации элементов.
+     * @return стрим
+     */
+    @SuppressWarnings("unchecked")
+    public <R> Streams<R> transform(Function<? super T, ? extends R> object) {
+
+        List<R> temp = new ArrayList<>();
+
+        suppliers.add(() -> {
+            temp.addAll(elements.stream()
+                    .map(object)
+                    .collect(Collectors.toList()));
+            return (List<T>) temp;
+        });
+
+        return (Streams<R>) this;
+    }
+
+    /**
+     * Сортировка элементов. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Данный метод должен уметь сортировать элементы по заданному правилу.
+     * <p>
```

Comment:
&lt;p&gt; tag should be placed immediately before the first word, with no space after.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a167605658>*